framework/CodeInterfaces/MELCOR/MELCORdata.py

```diff
@@ -56,6 +63,44 @@ def getTimeBlocks(self):
       timeBlock[info[1]] = self.lines[info[0]+1:endLineCnt]
     return timeBlock
 
+  def returnControlFunctions(self, timeBlock):
+    """
+      CONTROL FUNCTIONS EDIT
+      @ In, timeBlock, dict, {"time":[lines Of Output for that time]}
+      @ Out, functionValuesForEachTime, dict, {"time":{"functionName":"functionValue"}}
+    """
+    functionValuesForEachTime = {}
+    timeOneRegex_name = re.compile("^\s*CONTROL\s+FUNCTION\s+(?P<name>[^\(]*)\s+(\(.*\))?\s*IS\s+.+\s+TYPE.*$")
+    timeOneRegex_value = re.compile("^\s*VALUE\s+=\s+(?P<value>[^\s]*)")
+    startRegex = re.compile("\s*CONTROL\s*FUNCTION\s*NUMBER\s*CURRENT\s*VALUE")
+    regex = re.compile("^\s*(?P<name>( ?([0-9a-zA-Z-]+))*)\s+([0-9]+)\s*(?P<value>((([0-9.-]+)E(\+|-)[0-9][0-9])|((T|F))))\s*.*$")
+    for time,listOfLines in timeBlock.items():
+      functionValues = {}
+      start = -1
+      for lineNumber, line in enumerate(listOfLines):
+        if re.search(startRegex, line):
+          start = lineNumber + 1
+          break
+        elif re.search(timeOneRegex_name, line):
+          start = -2
+          break
+      if start > 0:
+        for lineNumber, line in enumerate(listOfLines[start:]):
+          if line.startswith(" END OF EDIT FOR CF"):
+            break
+          match = re.match(regex, line)
+          if match is not None:
+    	  	functionValues[match.groupdict()["name"]] = match.groupdict()["value"]
+      elif start == -2:
+      	for lineNumber, line in enumerate(listOfLines):
+      		fcnName = re.match(timeOneRegex_name, line)
```

Comment:
This indentation does not follow the indentation standards (we use 2 spaces and here I count 6-7).